src/middleware.ts

```diff
@@ -360,6 +361,15 @@ export type PersistOptions<
    */
   merge?: (persistedState: any, currentState: S) => S
 }
+interface PersistApi<S extends State> extends StoreApi<S> {
```

Comment:
```suggestion
export type StoreApiWithPersist<S extends State> = StoreApi<S> & {
```

Let's make it consistent with others. (tbh, I'm not sure whether type alias or interface is better in this case, or if it's just a preference thing. maybe, I can open up another PR to make everything type alias.)